gender_analysis/common.py

```diff
@@ -111,124 +112,50 @@
 # TODO(elsa): Investigate doctest errors in this file, may be a result of
 # my own system, not actual code errors
 
+def load_csv_to_dict(file_path):
+    """
+    Loads a csv file
+
+    :param file_path: can be a string or Path object
+    :return: a list of strings
+    """
+    if isinstance(file_path, str):
+        file_path = Path(file_path)
 
-class FileLoaderMixin:
-    """ The FileLoaderMixin loads files either locally or
-    remotely from Github (if run from an ipython notebook)
+    file_type = file_path.parts[-1][file_path.parts[-1].rfind('.'):]
+    file = open(file_path, encoding='utf-8')
 
-    Currently supported filetypes are: .csv, .txt
+    if file_type != '.csv':
```

Comment:
Do this check and throw this exception before opening the file handle. (i.e., no need to open the file for reading if the file_type is wrong!)